### PR TITLE
Edit customer in Stripe button is present in Edit Member page despite…

### LINKED ISSUE #2986: Edit customer in Stripe button is present in Edit Member page despite the active subscription belongs to a different gateway.
**Describe the bug**
Edit customer in Stripe button is present in Edit Member page despite the active subscription belongs to a different gateway.

**To Reproduce**

1. Setup Stripe as a payment gateway. 
2. Given a recurring level membership, checkout that level with subscriber user. 
3. Cancel that membership. 
4. Setup Authorize.net as a payment gateway 
5. Checkout a new membeship.
6. Observe the described bug

**Screenshots**
image

**Expected behavior**
Edit custome in Stripe button shouldn't be present if another  gateway membership is active

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only Paid Memberships Pro plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Memberlite](https://wordpress.org/themes/memberlite/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
WP 6.4.3
PMPro: dev branch
```
</details>

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
… the active subscription belongs to a different gateway.

 * Check if there's an active Stripe subscription before render the button
Screenshot 2024-05-13 at 3 20 59 PM
Screenshot 2024-05-13 at 3 19 34 PM

### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Check if there's an active Stripe subscription before render the button

Resolves #2986 

### How to test the changes in this Pull Request:

Follow steps from the linked issue.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Comment by dparker1005** [comment]
Is it still helpful to show a link to the customer in Stripe even if they don't have an active subscription? For example, if we want to look at their old subscriptions in Stripe for some reason?

**Comment by kimcoleman** [comment]
I could go either way on this but my gut says if they do not have an active Stripe subscription, do not show it. It could imply that there is an active connection there that isn't.

An alternative is to add the Edit Customer in Stripe button on the single Subscription admin view. This would then be present for active subs as well as cancelled.

File: adminpages/member-edit/pmpro-class-member-edit-panel-subscriptions.php [hunk 1]
```diff
@@ -12,10 +12,13 @@ public function __construct() {
 		$user = self::get_user();
 		$stripe = new PMProGateway_Stripe();
 		$customer = $stripe->get_customer_for_user( $user->ID );
-
+		$subscriptions = PMPro_Subscription::get_subscriptions_for_user( $user->ID );
+		$user_has_active_stripe_subscription = array_reduce( $subscriptions, function ( $carry, $subscription ) {
```
Comment (ideadude) [inline adminpages/member-edit/pmpro-class-member-edit-panel-subscriptions.php:16]: We can probably come up with a more readable/cleaner way to do this.

**Review by ideadude — COMMENTED** [review]
Just a comment to make this code clearer if we decide to do this.

**Comment by ideadude** [comment]
So tough.

It's a good idea to have a button on the individual subscription pages that links to those subscriptions in Stripe.

For the main "Edit customer in Stripe" button, we could move this to the user info tab and put it next to Edit User button. We should also make the text title case in the button IMO.

**Comment by dparker1005** [comment]
I just applied the recent suggestions from @ideadude.

@MaximilianoRicoTabo @kimcoleman Can you both give this new version a quick look to make sure that you don't see any issues?

**Comment by MaximilianoRicoTabo** [comment]
image

**Review by kimcoleman — APPROVED** [review]

**Pull request merged**